diff -Nuar moonlander_compromise-version_source/keymap.c ../qmk_firmware/keyboards/moonlander/keymaps/tutolmin/keymap.c
--- moonlander_compromise-version_source/keymap.c	2023-05-18 02:57:40.000000000 +0300
+++ ../qmk_firmware/keyboards/moonlander/keymaps/tutolmin/keymap.c	2023-05-18 14:40:23.276397265 +0300
@@ -186,6 +186,27 @@
     }
     break;
 
+    case TO(0): // this runs when TO(ENG) is pressed (the key to toggle layer 0 on)
+    case TG(2): // this runs when TG(COL) is pressed (the key to toggle layer 2 on)
+    if (record->event.pressed) {
+        register_code(KC_LCTRL);
+        register_code(KC_LSFT);
+        tap_code(KC_0);
+        unregister_code(KC_LSFT);
+        unregister_code(KC_LCTRL);
+    }
+    break;
+    
+    case TG(1): // this runs when TG(RUS) is pressed (the key to toggle layer 1 on)
+    if (record->event.pressed) {
+        register_code(KC_LALT);
+        register_code(KC_LSFT);
+        tap_code(KC_0);
+        unregister_code(KC_LSFT);
+        unregister_code(KC_LALT);
+    }
+    break;
+
     case RGB_SLD:
         if (rawhid_state.rgb_control) {
             return false;
