--- /media/torrents/moonlander/keymap.c	2023-06-01 07:11:52.000000000 +0300
+++ qmk_firmware/keyboards/moonlander/keymaps/tutolmin/keymap.c	2023-06-02 05:50:47.187075711 +0300
@@ -212,7 +212,9 @@
     break;
     case ST_MACRO_1:
     if (record->event.pressed) {
+      clear_oneshot_mods();
       SEND_STRING(SS_TAP(X_DOT) SS_DELAY(100) SS_TAP(X_SPACE));
+      set_oneshot_mods(MOD_LSFT);
     }
     break;
     case ST_MACRO_2:
@@ -222,7 +224,9 @@
     break;
     case ST_MACRO_3:
     if (record->event.pressed) {
+      clear_oneshot_mods();
       SEND_STRING(SS_TAP(X_7) SS_DELAY(100) SS_TAP(X_SPACE));
+      set_oneshot_mods(MOD_LSFT);
     }
     break;
     case ST_MACRO_4:
@@ -231,6 +235,27 @@
     }
     break;
 
+    case TO(0): // this runs when TO(ENG) is pressed (the key to toggle layer 0 on)
+    case TO(2): // this runs when TO(COL) is pressed (the key to toggle layer 2 on)
+    if (record->event.pressed) {
+        register_code(KC_LCTRL);
+        register_code(KC_LSFT);
+        tap_code(KC_0);
+        unregister_code(KC_LSFT);
+        unregister_code(KC_LCTRL);
+    }
+    break;
+    
+    case TO(1): // this runs when TO(RUS) is pressed (the key to toggle layer 1 on)
+    if (record->event.pressed) {
+        register_code(KC_LALT);
+        register_code(KC_LSFT);
+        tap_code(KC_0);
+        unregister_code(KC_LSFT);
+        unregister_code(KC_LALT);
+    }
+    break;
+
     case RGB_SLD:
         if (rawhid_state.rgb_control) {
             return false;
