--- keyboards/moonlander/keymaps/rututolmin/keymap.c.bak	2023-06-13 17:06:48.980092455 +0300
+++ keyboards/moonlander/keymaps/rututolmin/keymap.c	2023-06-13 18:43:26.545386520 +0300
@@ -1,5 +1,6 @@
 #include QMK_KEYBOARD_H
 #include "version.h"
+/*
 #include "keymap_german.h"
 #include "keymap_nordic.h"
 #include "keymap_french.h"
@@ -29,6 +30,8 @@
 #include "keymap_croatian.h"
 #include "keymap_turkish_q.h"
 #include "keymap_slovak.h"
+*/
+#include "keymap_ru_type.h"
 
 #define KC_MAC_UNDO LGUI(KC_Z)
 #define KC_MAC_CUT LGUI(KC_X)
@@ -242,6 +245,13 @@
     }
     break;
 
+    case RU_MINS: // - in Russian should NOT be shifted
+    case RU_DQUO: // \" in Russian should NOT be shifted
+    if (biton32(layer_state) == 1) {
+      clear_oneshot_mods();
+    }
+    break;
+
     case TO(0): // this runs when TO(ENG) is pressed (the key to toggle layer 0 on)
     case TO(2): // this runs when TO(COL) is pressed (the key to toggle layer 2 on)
     if (record->event.pressed) {
